Decisions.ExperianServices/ExperianInitializer.cs

```diff
@@ -0,0 +1,84 @@
+                            "entity_folder.folder_id",
+                            QueryMatchType.Equals)
+                    })
+                ));
+
+            compositeSelectStatement.WhereConditions.WhereConditions.Add(
+                new AndWhereSet(new WhereCondition[]
+                {
+                    new FieldWhereCondition(
+                        "element_registration.entity_name",
+                        QueryMatchType.EqualsWithoutCase,
+                        "Experian Consumer Credit Profile"),
+                    new FieldWhereCondition(
+                        "entity_folder.full_path",
+                        QueryMatchType.EqualsWithoutCase,
+                        "System/Designers/Catalog/Credit Profile/Experian Steps")
+                }));
+
+          
+            DynamicORM orm = new DynamicORM();
+
+            DataSet results = orm.RunQuery(compositeSelectStatement);
+
+            /*
+             * We only need to do this if the old step categories are there.
+             */
+            if (results != null && results.Tables[0].Rows.Count > 0)
+            {
+                string sqlQuery = @"
+                                IF EXISTS (SELECT 1 FROM sys.triggers  WHERE Name = 'Delete_element_registration')
+                                    DISABLE TRIGGER Delete_element_registration ON element_registration;  
+
+                                DELETE dbo.element_registration
+                                FROM dbo.element_registration
+                                INNER JOIN dbo.entity_folder ON element_registration.entity_folder_id = entity_folder.folder_id
+                                WHERE element_registration.entity_name = 'Experian Consumer Credit Profile'
+                                AND entity_folder.full_path = 'System/Designers/Catalog/Credit Profile/Experian Steps';
+
+                                IF EXISTS (SELECT 1 FROM sys.triggers  WHERE Name = 'Delete_element_registration')
+                                   ENABLE TRIGGER Delete_element_registration ON element_registration; 
+                                   
+                                /* As we disabled the Trigger when Deleting the Element Registrations, we need to manually remove these items from the Entity Header Data */   
+                                IF EXISTS (SELECT 1 FROM sys.triggers  WHERE Name = 'entity_header_data_delete_trigger')
+                                   DISABLE TRIGGER entity_header_data_delete_trigger ON entity_header_data;  
+
+                                delete from entity_header_data where id in (
+                                select ehd.id from entity_header_data ehd left join element_registration er 
+                                on ehd.id = er.component_registration_id
+                                where er.component_registration_id is null and entity_type_name = 'DecisionsFramework.ServiceLayer.Services.ConfigurationStorage.ElementRegistration')
```

Comment:
Double checking this query is correct. Seems broad unless this null is triggered by the first query.

Reply:
I took this from C:\workspaces\decisions-platform-trunk\modules\Decisions.Core\SQLScripts\MSSQL\delete_old_step_categories.sql 

We're doing the same thing there in order to move steps. I think it's because the item is already removed from the element_registration table.